packages/reflect-yjs/src/provider.ts

```diff
@@ -23,33 +22,25 @@ export class Provider {
     ydoc.on('update', this.#handleUpdate);
     ydoc.on('destroy', this.#handleDestroy);
 
-    this.#cancelUpdateSubscribe = reflect.subscribe(
+    this.#cancelSubscribe = reflect.subscribe(
       async tx => {
-        let v = await tx.get(yjsProviderClientKey(this.name));
-        if (typeof v !== 'string') {
-          v = await tx.get(yjsProviderServerKey(this.name));
-        }
-        return typeof v === 'string' ? v : null;
-      },
-      docStateFromReflect => {
-        if (docStateFromReflect !== null) {
-          const update = base64.toByteArray(docStateFromReflect);
-          Y.applyUpdateV2(ydoc, update);
-        }
-      },
-    );
-
-    this.#cancelVectorSubscribe = reflect.subscribe(
-      async tx => {
-        const v = await tx.get(yjsProviderServerKey(this.name));
-        return typeof v === 'string' ? v : null;
+        const serverUpdate = await getServerUpdate(this.name, tx);
+        return {
+          serverUpdate,
+          clientOverServerUpdate:
+            (await await getClientUpdate(this.name, tx)) ?? serverUpdate,
```

Comment:
```suggestion
            (await getClientUpdate(this.name, tx)) ?? serverUpdate,
```

Reply:
I would use an array here because it "compiles" better

Reply:
Also, maybe skip `?? serverUpdate` and deal with that in the `onDone`?

Reply:
done